### PR TITLE
#4077: Add e2e Profiles test

### LINKED ISSUE #4077: Profile & Kfam: E2E tests
Profile & Kfam: E2E tests

### PR DESCRIPTION
**Pull request by nrchakradhar**
Partially fixes #4077 
Adding e2e Profiles test.
Made some minor changes in profile_controller.go for spelling errors.

- [x] After verification in CD pipeline, need to uncomment exception raising.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/4455)

### CODE REVIEW

**Comment by nrchakradhar** [comment]
@jlewi @kunmingg @lluunn 
Can one of you help with adding /ok-to-test?

**Comment by nrchakradhar** [comment]
@krishnadurai 
> /ok-to-test

Thanks. I have updated the PR and pre-submit is passing.

File: kubeflow/profiles/tests/profiles_test.py [hunk 1]
```diff
@@ -65,19 +142,61 @@ def test_profiles():
...
+      body=wf_result)
+    logging.info("Profile created:\n%s", yaml.safe_dump(resp))
+  # Profiles status can be one of Succeeded, Failed, Unknown
+  # TODO: check if status comes by using callbacks.
+  time.sleep(20)
+  return group, version, name
+
+def test_profiles(record_xml_attribute, profileFile= "profile_v1beta1_profile.yaml"):
+  util.set_pytest_junit(record_xml_attribute, "test_profile_e2e")
+  app_credentials = os.getenv("GOOGLE_APPLICATION_CREDENTIALS")
```
Comment (jlewi) [inline kubeflow/profiles/tests/profiles_test.py]: Use util.maybe_activate_service_account

Comment (nrchakradhar) [reply kubeflow/profiles/tests/profiles_test.py]: @jlewi Done. PTAL

**Comment by jlewi** [comment]
@nrchakradhar please send PR to add yourself to
https://github.com/kubeflow/internal-acls/blob/master/github-orgs/kubeflow/org.yaml

To join the org.

**Comment by nrchakradhar** [comment]
The failure seems to be not related to any changes done in this PR. 

WARNING: fetch http://nl.alpinelinux.org/alpine/edge/main/x86_64/APKINDEX.tar.gz
Ignoring http://nl.alpinelinux.org/alpine/edge/community/x86_64/APKINDEX.tar.gz: No such file or directory
WARNING: Ignoring http://nl.alpinelinux.org/alpine/edge/main/x86_64/APKINDEX.tar.gz: No such file or directory
WARNING: The repository tag for world dependency 'libstdc++@edge' does not exist
WARNING: The repository tag for world dependency 'chromium@edge' does not exist
WARNING: The repository tag for world dependency 'nss@edge' does not exist
ERROR: (gcloud.builds.submit) build 0c894322-4dee-43c6-b06c-0523951e840c completed with status "FAILURE"
WARNING: The repository tag for world dependency 'freetype@edge' does not exist
WARNING: The repository tag for world dependency 'harfbuzz@edge' does not exist
WARNING: The repository tag for world dependency 'ttf-freefont@edge' does not exist
ERROR: Not committing changes due to missing repository tags. Use --force-broken-world to override.
The command '/bin/sh -c apk update && apk upgrade &&     echo @edge http://nl.alpinelinux.org/alpine/edge/community >>
/etc/apk/repositories &&     echo @edge http://nl.alpinelinux.org/alpine/edge/main >> /etc/apk/repositories &&     apk
add --no-cache bash chromium@edge nss@edge     freetype@edge     harfbuzz@edge     ttf-freefont@edge     libstdc++@edge
' returned a non-zero code: 99
ERROR
ERROR: build step 0 "gcr.io/cloud-builders/docker" failed: exit status 99

**Pull request merged**